tools/PI/DevHome.PI/Helpers/ExternalTool.cs

```diff
@@ -156,14 +185,21 @@ internal async override void InvokeTool(Window? parentWindow, int? targetProcess
                 // false otherwise. However, if there's no registered app for the protocol, it shows
                 // the "get an app from the store" dialog, and returns true. So we can't rely on the
                 // return value to know if the tool was actually launched.
-                var result = await Launcher.LaunchUriAsync(new Uri(parsedArguments));
+                var result = await Launcher.LaunchUriAsync(new Uri(commandLineArgs));
                 if (result != true)
                 {
                     // We get here if the user supplied a valid registered protocol, but the app failed to launch.
                     var errorMessage = string.Format(
-                        CultureInfo.InvariantCulture, _errorMessageText, parsedArguments);
+                        CultureInfo.InvariantCulture, _errorMessageText, commandLineArgs);
                     throw new InvalidOperationException(errorMessage);
                 }
+
+                // Currently we can't get the process object of the launched app via LaunchUriAsync. If we
+                // ever do and want to populate ToolLaunchOptions.LaunchedProcess, we'll need to revisit the async behavior here,
+                // as when we return from this function (and our async operation is still ongoing) our callers currently expect to
+                // be able to read ToolLaunchOptions.LaunchedProcess immedately... there isn't an indication that they would need to
+                // do an await first.
+                // process = ProcessLaunchedFromLaunchUriAsync;

```

Comment:
is `process = ProcessLaunchedFromLauncherUriAsync` part of the comment?

Reply:
Yep, that would be the dream line of code I would write if I could get the process. I can understand how that could be confusing though. I'll remove it.